src/rustc_lexer/src/cursor.rs

```diff
@@ -0,0 +1,40 @@
+use std::str::Chars;
+
+pub(crate) struct Cursor<'a> {
+    chars: Chars<'a>,
+    prev: char,
+}
+
+pub(crate) const EOF_CHAR: char = '\0';
+
+
+impl Cursor<'_> {
+    pub(crate) fn new(input: &str) -> Cursor {
+        Cursor {
+            chars: input.chars(),
+            prev: EOF_CHAR,
+        }
+    }
+    // for assertions only
```

Comment:
Could you perhaps use `debug_assert!` and `#[cfg(debug_assertions)]` to make `Cursor` contain just `Chars` when debug assertions are off?
I suspect these assertions are sanity checks and not necessary for correct behavior.

Reply:
Yeah, I intend to flip all asserts to debug_asserts in the end, but it’s useful to have them always on while developing

Reply:
I mean, you can just turn on all debug assertions always (in `config.toml`, or, if you're building just this crate separately, in [`Cargo.toml`'s `[profile.release]` section](https://doc.rust-lang.org/cargo/reference/manifest.html#the-profile-sections) - assuming you're not using debug mode, which just has these on).
This also gives you `debug!` logging, which is handy to have.

Reply:
It's just that I have to Cargo.toml (this crate's and the root one), and I sincerely have no idea if `./x.py` uses debug assertions :) So it's easier to do something I know will work

Reply:
`./x.py` uses what's set in `config.toml` (i.e. set `debug-assertions = true` in it)